### ISSUE TITLE
[BUG] `DTypeCastTransform` changes arbitrary keys

### ISSUE THREAD

**Issue by matteobettini** [pytorch/rl#2085]
I create a transform `transform = DTypeCastTransform(dtype_in=torch.uint8, dtype_out=torch.float32, in_keys=["image"])` to change the "image" into float and my float "reward" is changed to int instead.

```python
import torch
from tensordict import TensorDict

from torchrl.data import TensorDictReplayBuffer, LazyTensorStorage, RandomSampler
from torchrl.envs import DTypeCastTransform

if __name__ == "__main__":
    td = TensorDict(
        {
            "image": torch.zeros(1, dtype=torch.uint8),
            "reward": torch.zeros(1, dtype=torch.float32),
        },
        batch_size=[1],
    )
    transform = DTypeCastTransform(
        dtype_in=torch.uint8, dtype_out=torch.float32, in_keys=["image"]
    )

    buffer = TensorDictReplayBuffer(
        storage=LazyTensorStorage(10),
        sampler=RandomSampler(),
        batch_size=1,
        transform=transform,
    )

    buffer.extend(td)
    sample_td = buffer.sample()

    print(sample_td.get("reward").dtype)  # uint
    assert sample_td.get("reward").dtype == torch.float32

```

**Comment by vmoens** [comment]
It isn't a bug, it's because you didn't pass the inv_key set
```python
    transform = DTypeCastTransform(
        dtype_in=torch.uint8, dtype_out=torch.float32, in_keys=["image"], in_keys_inv=[],
    )
```
works fine. I will re-write the doc to make this explicit. 
The context is that if you don't say what you want as in_keys_inv we assume everything that is float32 will be uint8.

A possible patch for this when it's not intuitive would be that the default for `in_keys_inv` is `[]` and if you want them all you set it to `None` but that would be bc-breaking so at most we can do that in a couple of releases.

Another possible fix is that if you pass anything like in_keys or in_keys_inv we assume you passed everything you wanted. That is less BC-breaking than the above and I'd be willing to patch that as is.

**Comment by vmoens** [comment]
BTW this warning was already there now that I look at it
```
            warnings.warn(
                "in_keys have been provided but not in_keys_inv. From v0.5, "
                "this will result in in_keys_inv being an empty list whereas "
                "now the input keys are retrieved automatically. "
                "To silence this warning, pass the (possibly empty) "
                "list of in_keys_inv.",
                category=DeprecationWarning,
            )
```

**Comment by matteobettini** [comment]
I would change the docstring though

>  * If the keys are provided, those entries and those entries only will be
>         transformed from ``dtype_in`` to ``dtype_out`` entries;

The content of that warning (which somehow i cannot get to show up) should be in the docstring too

**Comment by vmoens** [comment]
Sure feel free to propose changes you see fit
